libqf/libqfcore/src/core/string.h

```diff
@@ -17,6 +17,11 @@ class QFCORE_DECL_EXPORT String : public QString
 	String() : QString() {}
 	String(const QString &str) : QString(str) {}
 	String(const String &str) : QString(str) {}
+	qf::core::String& operator=(const String &str) {
```

Comment:
this is base class implementation
```c++
String &QString::operator=(const QString &other) noexcept
{
    other.d->ref.ref();
    if (!d->ref.deref())
        Data::deallocate(d);
    d = other.d;
    return *this;
}
```
what is the gain of proposed change? Why throw away QString implicit sharing zero copy?

Reply:
compiler throws warnings because `The generation of the implicitly-defined copy assignment operator is deprecated if T has user-declared copy constructor. (since C++11)`  

this change introduces explicit copy assignment operator, so the build log is not cluttered with the warnings mentioned 

the implementation is not *nice* because the QString class is not meant to be inherited in the first place so only public functions are being used. there might be better way to implement this

Reply:
Rather than introducing another explicit definition, you probably only need to *explicitly* default this operation to resolve this warning:
~~~
String& operator=(const String &str) = default;
~~~
In this class, this recommendation would proably apply to all special member function. Or just remove them all (rule of zero).

Reply:
exactly

Reply:
```c++
	String() : QString() {}
	String(const QString &str) : QString(str) {}
```
previous explicit ctors decl might be IMO removed as well

Reply:
Rule of zero probably doesn't work because there is one custom constructor (from QString). So the price for this convenience class (adding members to QString) is explicit (defaulted) special member functions. 
(The alternative is to stick with plain QString, and use free functions.)

Reply:
As I understand it, this should be the minimal implementation.
```cpp
String() = default;
String(const QString &str) : QString(str) {}
```